src/java.base/share/classes/sun/security/rsa/RSASignature.java

```diff
@@ -257,15 +259,17 @@ public static byte[] decodeSignature(ObjectIdentifier oid, byte[] sig)
             throw new IOException("SEQUENCE length error");
         }
         AlgorithmId algId = AlgorithmId.parse(values[0]);
-        if (algId.getOID().equals(oid) == false) {
+        if (!algId.getOID().equals(oid)) {
             throw new IOException("ObjectIdentifier mismatch: "
                 + algId.getOID());
         }
         if (algId.getEncodedParams() != null) {
             throw new IOException("Unexpected AlgorithmId parameters");
         }
-        byte[] digest = values[1].getOctetString();
-        return digest;
+        if (values[1].isConstructed()) {
```

Comment:
What is the source of these changes? I'm having trouble finding them in the jdk21u backport or the original change in tip.

Reply:
This change is from https://github.com/openjdk/jdk21u/commit/14e7d911997d33eba2893991fa0e2f507aa977f8 and should not have been included when resolving the conflicts.